packages/flutter/lib/src/material/bottom_app_bar.dart

```diff
@@ -0,0 +1,135 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material.dart';
+import 'scaffold.dart';
+
+// Examples can assume:
+// Widget bottomAppBarContents;
+
+/// A container that shows in the [Scaffold.bottomNavigationBar] slot, and can
+/// have a notch in which a [FloatingActionButton] is showing.
+///
+/// Typically used with a [Scaffold] and a [FloatingActionButton].
+///
+/// ## Sample code
+///
+/// ```dart
+/// new Scaffold(
+///   bottomNavigationBar: new BottomAppBar(
+///     color: Colors.white,
+///     child: bottomAppBarContents,
+///   ),
+///   floatingActionButton: new FloatingActionButton(onPressed: null),
+/// )
+/// ```
+///
+/// See also:
+///
+///   * [NotchMaker] a closure used for creating a notch in a shape.
+///   * [ScaffoldGeometry.floatingActionBarNotchMaker] the [NotchMaker] used to
+///   make a notch for the [FloatingActionButton]
+///   * [FloatingActionButton] which the [BottomAppBar] makes a notch for.
```

Comment:
nit: for consistency with other dartdocs, only indent the bullets by one (asterisks aligned under the first "e") and wrap the text of these bullets so that they line up with the first character of their first line.

Reply:
done